nativescript-angular/router/ns-location-strategy.ts

```diff
@@ -554,26 +563,33 @@ export class NSLocationStrategy extends LocationStrategy {
         return pathToOutlet || lastPath;
     }
 
-    findOutletByModal(modalNavigation: number, isShowingModal?: boolean): Outlet {
-        return this.outlets.find((outlet) => {
-            let isEqual = outlet.modalNavigationDepth === modalNavigation;
-            return isShowingModal ? isEqual && outlet.showingModal : isEqual;
-        });
-    }
-
     findOutlet(outletKey: string, activatedRouteSnapshot?: ActivatedRouteSnapshot): Outlet {
-        let outlet: Outlet = this.outlets.find((currentOutlet) => currentOutlet.outletKeys.indexOf(outletKey) > -1);
+        const that = this;
+        let outlet: Outlet = this.outlets.find((currentOutlet) => {
+            let equalModalDepth = currentOutlet.modalNavigationDepth === that._modalNavigationDepth;
+            return currentOutlet.outletKeys.indexOf(outletKey) > -1 && equalModalDepth;
```

Comment:
Move the `equalModalDepth` check to the front. It's already evaluated and if it's `false` the `indexOf` won't evaluate at all.